syntax = 'proto3';

import "face_common.proto";

message FaceAlignmentHeader {
    repeated BoundingBox   source_bboxes = 1;
    // DLib can't currently allow arbitrary alignments
    //repeated FaceLandmarks source = 2;
    //FaceLandmarks target = 3;
}

message FaceAlignmentRequest {
    oneof header_or_chunk {
        FaceAlignmentHeader header = 1;
        ImageRGB image_chunk = 2;
    }
}

message FaceAlignmentResponseHeader {

}

message FaceAlignmentResponse {
    oneof header_or_chunk {
        FaceAlignmentResponseHeader header = 1;
        ImageRGB image_chunk = 2;
    }
}

service FaceAlignment {
    rpc AlignFace (stream FaceAlignmentRequest) returns (stream FaceAlignmentResponse) {};
}